nova/virt/vmwareapi/vmops.py

```diff
@@ -1535,6 +1516,20 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                 self._session._call_method(self._session.vim,
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
+
+                with lockutils.lock(server_group_info.uuid,
+                                lock_file_prefix='nova-vmware-server-group'):
```

Comment:
Did you test this with a VM that's not in a server-group? Because I just did and the tests work. But if I add one line self.assertTrue(False)` to `nova.tests.unit.virt.vmwareapi.test_driver_api.VMwareAPIVMTestCase.test_destroy$` I can see the output the test was hiding. Nothing works as expected in there ... because that Exception is catched in the un-registering-error-handler ...